packages/editor/core/src/lib/editor-commands.ts

```diff
@@ -34,32 +35,77 @@ export const toggleUnderline = (editor: Editor, range?: Range) => {
   else editor.chain().focus().toggleUnderline().run();
 };
 
-export const toggleCodeBlock = (editor: Editor, range?: Range) => {
-  // Check if code block is active then toggle code block
-  if (editor.isActive("codeBlock")) {
-    if (range) {
-      editor.chain().focus().deleteRange(range).toggleCodeBlock().run();
-      return;
+const replaceCodeBlockWithContent = (editor: Editor) => {
+  try {
+    const { schema, tr } = editor.state;
+    const { codeBlock } = schema.nodes;
+    let replaced = false;
+
+    const replaceCodeBlock = (from: number, to: number, textContent: string) => {
+      // Validate range
+      const docSize = editor.state.doc.content.size;
+      if (from < 0 || to > docSize || from > to) {
+        console.error("Invalid range for replacement.");
+        return;
+      }
+
+      const lines = textContent.split(/\r?\n/);
+      const parsedContent = lines.map((line) => `<p>${line.replace(/ /g, "&nbsp;")}</p>`).join("");
+      const div = document.createElement("div");
+      div.innerHTML = parsedContent;
```

Comment:
I think you'll need to Sanitize the pasted block for scripts before adding it to the innerHTML. Otherwise it will be a vulnerability

Reply:
Thanks to your suggestions, I've removed the innerHTML method, instead parse it line by line to create paragraph nodes via prosemirror apis using the parsed content seperated by \n